### PR TITLE
AddApptCommand

### LINKED ISSUE #78: Add appointment feature
- [x] AddApptCommand

- [ ] ViewApptCommand

- [ ] DeleteApptCommand

### PR DESCRIPTION
**Pull request by jeffypie369**
Not all tests covered yet
Partially resolves #78

### CODE REVIEW

**Comment by jeffypie369** [comment]
All tests covered.

File: src/main/java/seedu/address/storage/XmlAdaptedAppointment.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * JAXB-friendly adapted version of the Appointment
+ */
+
+@XmlRootElement
+public class XmlAdaptedAppointment {
+
+    @XmlElement(required = true)
+    private Type type;
```
Comment (snajef) [inline src/main/java/seedu/address/storage/XmlAdaptedAppointment.java]: You'll need to change this to an XmlAdapted equivalent, else the XML representation of your object will not store Type data.

File: src/test/java/seedu/address/logic/commands/AddApptCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        appt = new Appointment(type, procedure, dateTime, doctor);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void constructor_nullNric_throwsNullPointerEception() {
+        Command command = new AddApptCommand(null, appt);
+    }
+
+    @Test
+    public void execute_addapptToExistingPatient_addapptSuccessful() throws CommandException {
```
Comment (snajef) [inline src/test/java/seedu/address/logic/commands/AddApptCommandTest.java]: Consider camelCasing your test method name here e.g. execute_addApptToExistingPatient_addApptSuccessful()

File: src/test/java/seedu/address/logic/commands/AddApptCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        ModelStubAcceptingAddappt modelStub = new ModelStubAcceptingAddappt(patient);
```
Comment (snajef) [inline src/test/java/seedu/address/logic/commands/AddApptCommandTest.java]: Consider consistency in camelCasing e.g. AddAppt above, but here it is Addappt

File: src/test/java/seedu/address/logic/commands/AddApptCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        CommandResult commandResult = new AddApptCommand(patient.getNric(), appt).execute(modelStub,
+                commandHistory);
+        assertEquals(String.format(AddApptCommand.MESSAGE_SUCCESS, patient.getNric()), commandResult.feedbackToUser);
+    }
+
+    @Test
+    public void execute_addapptToNonexistentPatient_throwsCommandException() throws Exception {
```
Comment (snajef) [inline src/test/java/seedu/address/logic/commands/AddApptCommandTest.java]: Same issue with naming consistency,  see above comment.

File: src/test/java/seedu/address/logic/commands/AddApptCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        thrown.expect(CommandException.class);
+        thrown.expectMessage(AddApptCommand.MESSAGE_NO_SUCH_PATIENT);
+        addApptCommand.execute(modelStub, commandHistory);
+    }
+
+    @Test
+    public void equals() {
+        Person alice = new PersonBuilder().withName("Alice").build();
+        Person bob = new PersonBuilder().withNric("S2345678B").withName("Bob").build();
+        Appointment diffAppt = new Appointment(type, "Heat Bypass", dateTime, doctor);
```
Comment (snajef) [inline src/test/java/seedu/address/logic/commands/AddApptCommandTest.java]: Misspelling of "Heart Bypass".

**Pull request merged**